storagemarket/deal_execution.go

```diff
@@ -146,6 +147,30 @@ func (p *Provider) transferAndVerify(ctx context.Context, pub event.Emitter, dea
 	return p.updateCheckpoint(ctx, pub, deal, dealcheckpoints.Transferred)
 }
 
+func (p *Provider) waitForTransferFinish(ctx context.Context, handler transport.Handler, pub event.Emitter, deal *types.ProviderDealState) error {
+	defer handler.Close()
+
+loop:
+	for {
+		select {
+		case evt, ok := <-handler.Sub():
+			if !ok {
+				break loop
+			}
+			if evt.Error != nil {
+				return fmt.Errorf("data-transfer failed: %w", evt.Error)
```

Comment:
nit: you already moved "data-transfer failed" to the calling code so I think you can just directly return the error here and on line 167 below

Reply:
done.